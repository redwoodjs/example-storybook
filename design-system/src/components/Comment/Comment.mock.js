export const standard = () => {
  return {
    comment: {
      name: "April O'Neil",
      body: 'Cowabunga dudes. This is the first comment!',
      createdAt: '2022-04-01T12:34:56Z',
    },
  }
}
