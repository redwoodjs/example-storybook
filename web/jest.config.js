// More info at https://redwoodjs.com/docs/project-configuration-dev-test-build

const config = {
  rootDir: '../',
  preset: '@redwoodjs/testing/config/jest/web',
  verbose: true,
  moduleNameMapper: {
    // mock redwood magic webpack aliases
    // see: https://github.com/redwoodjs/redwood/blob/main/packages/testing/config/storybook/preview.js
    '~__REDWOOD__USER_WEB_DEFAULT_CSS': 'identity-obj-proxy',
    '~__REDWOOD__USER_STORYBOOK_PREVIEW_CONFIG':
      '<rootDir>/web/config/storybook.preview.js',
  },
  transform: {
    '^.+\\.mdx?$': '@storybook/addon-docs/jest-transform-mdx', // https://github.com/storybookjs/storybook/issues/11376#issuecomment-757417250
  },
  setupFilesAfterEnv: [`<rootDir>/web/test/mocks.js`],
}

module.exports = config
