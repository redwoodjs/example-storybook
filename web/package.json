{
  "name": "web",
  "version": "0.0.0",
  "private": true,
  "browserslist": {
    "development": [
      "last 1 version"
    ],
    "production": [
      "defaults",
      "not IE 11",
      "not IE_Mob 11"
    ]
  },
  "dependencies": {
    "@redwoodjs/forms": "0.49.0",
    "@redwoodjs/router": "0.49.0",
    "@redwoodjs/web": "0.49.0",
    "prop-types": "15.8.1",
    "react": "17.0.2",
    "react-dom": "17.0.2"
  }
}
