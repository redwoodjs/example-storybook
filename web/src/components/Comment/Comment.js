const Comment = () => {
  return (
    <div>
      <h2>{'Comment'}</h2>
      <p>{'Find me in ./web/src/components/Comment/Comment.js'}</p>
    </div>
  )
}

export default Comment
