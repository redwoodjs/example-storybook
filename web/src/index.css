/**
 * START --- SETUP TAILWINDCSS EDIT
 *
 * `yarn rw setup ui tailwindcss` placed these imports here
 * to inject Tailwind's styles into your CSS.
 * For more information, see: https://tailwindcss.com/docs/installation#include-tailwind-in-your-css
 */
@import 'tailwindcss/base';
@import 'tailwindcss/components';
@import 'tailwindcss/utilities';
/**
 * END --- SETUP TAILWINDCSS EDIT
 */
body {
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
}
ul {
  list-style-type: none;
  margin: 1rem 0;
  padding: 0;
}
li {
  display: inline-block;
  margin: 0 1rem 0 0 ;
}
h1 > a {
  text-decoration: none;
  color: black;
}
button, input, label, textarea {
  display: block;
  outline: none;
}
label {
  margin-top: 1rem;
}
.error {
  color: red;
}
input.error, textarea.error {
  border: 1px solid red;
}
.form-error {
  color: red;
  background-color: lavenderblush;
  padding: 1rem;
  display: inline-block;
}
.form-error ul {
  list-style-type: disc;
  margin: 1rem;
  padding: 1rem;
}
.form-error li {
  display: list-item;
}
.flex-between {
  display: flex;
  justify-content: space-between;
}
.flex-between button {
  display: inline;
}
