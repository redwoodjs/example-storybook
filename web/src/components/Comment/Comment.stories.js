import Comment from './Comment'

export const generated = (args) => {
  return <Comment {...args} />
}

export default { title: 'Components/Comment' }
