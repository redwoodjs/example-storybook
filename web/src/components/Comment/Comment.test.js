import { render } from '@redwoodjs/testing/web'

import Comment from './Comment'

//   Improve this test with help from the Redwood Testing Doc:
//    https://redwoodjs.com/docs/testing#testing-components

describe('Comment', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<Comment />)
    }).not.toThrow()
  })

  it('Comment fragment matches snapshot', () => {
    const { asFragment } = render(<Comment />)
    expect(asFragment()).toMatchSnapshot()
  })
})
