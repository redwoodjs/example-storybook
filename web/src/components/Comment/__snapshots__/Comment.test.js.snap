// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment Comment fragment matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Comment
    </h2>
    <p>
      Find me in ./web/src/components/Comment/Comment.js
    </p>
  </div>
</DocumentFragment>
`;
